sale_purchase_history/models/product_sale_history.py

```diff
@@ -0,0 +1,25 @@
+from odoo import api, fields, models
+
+class ProductSaleHistory(models.TransientModel):
+    _name = 'product.sale.history'
+
+    product_id = fields.Many2one('product.product')
+    is_sale = fields.Boolean(compute='_compute_active_model')
+    is_purchase = fields.Boolean(compute='_compute_active_model')
```

Comment:
rather than a compute, to reduce complexity and lines of code you could use defaults as this is a wizard, we don't care about maintaining the record

default=lambda self: self.env.context.get("something_from_context", False)